### PR TITLE
Enhancement - add support for sharefile oauth

### PR DESCRIPTION
**Pull request by sagart**

### CODE REVIEW

**Comment by sagart** [comment]
On hold, don't merge till we have some pending backend changes ready.

**Comment by sagart** [comment]
@crochik have refactored it to have a common base class OAuth.
@rgmills exposed the [ApiUrl for overriding](https://github.com/podio/podio-dotnet/blob/enhancement/AddSupportForSFIntegration/Source/Podio%20.NET/Podio.cs#L25)

File: Source/Podio .NET/Podio.cs [hunk 1]
```diff
@@ -207,10 +206,7 @@ private HttpRequestMessage CreateHttpRequest(string url, HttpMethod httpMethod,
 
             if (addAuthorizationHeader)
             {
-                if (OAuth != null && !string.IsNullOrEmpty(OAuth.AccessToken))
-                {
-                    request.Headers.Authorization = new AuthenticationHeaderValue("OAuth2", OAuth.AccessToken);
-                } 
+                OAuth.AddAuthorizationHeader(request.Headers);
```
Comment (crochik) [inline Source/Podio .NET/Podio.cs]: what if oauth is null ?

**Comment by sagart** [comment]
@crochik have fixed the pointed out NRE and double checked on such occurrences. Yes we are throwing authorization error appropriately now. Have squashed it.

**Pull request merged**